code/string_algorithms/src/morse_code/morse_translator.cpp

```diff
@@ -0,0 +1,132 @@
+#include <iostream>
+#include <map>
+
+using namespace std;
+
+string wordToMorse(string input, map<string,string> mp, map<string,string>::iterator it);
+string morseToWord(string input, map<string,string> mp, map<string,string>::iterator it);
+
+int main() 
+{
+    map <string,string> key;
```

Comment:
You do know that you can initialize the map at once, right?

```C++
std::map<std::string, std::string> key
{
    {"A", ".-"},
    {"B", "-..."},
    // ...
};
```

Also, you might want to consider using a `char` as the first key, which is a single character.

However, I feel that a better structure would be `std::vector<std::pair<char, std::string>>`. With this, you don't need to store character-to-morse _and_ morse-to-character in the same container. You can instead write standalone functions for doing the conversion.